crates/telio-proxy/src/proxy.rs

```diff
@@ -191,6 +243,40 @@ impl StateEgress {
 
         Ok(())
     }
+
+    fn log_error(err: &Error) {
+        match err {
+            Error::IoError(e) => telio_log_error!("Unable to send. Error {}", e.kind()),
+            _ => telio_log_error!("Unable to send. Error {}", err),
+        }
+    }
+
+    async fn handle_error(&mut self, err: Error) {
+        if self.conn_state != OutboundConnState::Error {
+            Self::log_error(&err);
+
+            match ErrorType::from(err) {
+                ErrorType::RecoverableError => (),
+                ErrorType::UnrecoverableError => {
+                    Self::sleep_forever().await;
+                }
+                ErrorType::SocketIOError => {
+                    if let Err(err) = SocketPool::new_udp(
+                        SocketAddr::from((Ipv4Addr::LOCALHOST, 0)),
+                        Some(UdpParams(SocketBufSizes {
+                            rx_buf_size: Some(SOCK_BUF_SZ),
+                            tx_buf_size: Some(SOCK_BUF_SZ),
+                        })),
+                    )
+                    .await
+                    {
+                        telio_log_error!("Cannot open socket. ERR :{}", err);
+                    };
+                }
+            }
+            self.conn_state = OutboundConnState::Error;
```

Comment:
If there's an `UnrecoverableError`, this line won't be reached as task is suspended. Move this assignment to the beginning of the function.